python/cugraph/structure/number_map.py

```diff
@@ -0,0 +1,263 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cugraph
+
+class NumberMap:
```

Comment:
Where does this object live in:
- single GPU pipeline?
- multi GPU pipeline?

Reply:
My goal was that the same object would be used in both pipelines.

Reply:
Yes but where in each pipeline

Reply:
On a single GPU, this object would be internal to the Graph object.  It would be automatically generated when from_cudf_edgelist is called, it would be used by all analytics when renumbering/unrenumbering capabilities are required.

If a user decided to access the renumbering features themselves (currently supported in single GPU) the output of the renumbering function would be this object instead of the current Dataframe that gets returned, so that the user can directly call the to_vertex_id and from_vertex_id methods as they need to.

I haven't seen a complete description of the multi GPU pipeline.  My assumption is that it will be similar to the single GPU pipeline.  We would have some overarching python graph object that would create and maintain this object.

Reply:
> On a single GPU, this object would be internal to the Graph object. It would be automatically generated when from_cudf_edgelist is called, it would be used by all analytics when renumbering/unrenumbering capabilities are required.

Sounds good.

> If a user decided to access the renumbering features themselves (currently supported in single GPU) the output of the renumbering function would be this object instead of the current Dataframe that gets returned, so that the user can directly call the to_vertex_id and from_vertex_id methods as they need to.

Ok. We would need to document this object and add an example in the user doc.

> I haven't seen a complete description of the multi GPU pipeline. 

There are GitHub Issues and PRs for OPG PageRank and BFS. Let me know if you have any specific questions. 

> My assumption is that it will be similar to the single GPU pipeline. We would have some overarching python graph object that would create and maintain this object.

To some extent. As of now, there is not one global multi GPU graph object. A distributed dask cudf edge list is accepted and single GPU Di/Graphs are created locally for each rank.

Reply:
@ChuckHastings I lost track if you added doc for accessing the renumbered data. 
This is motivated by users requests, see for instance #925. 
Feel free to resolve and comment with the commit id.